src/components/Microsoft.DotNet.UpgradeAssistant/TargetFramework/WindowsSdkTargetFrameworkSelectorFilter.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.DotNet.UpgradeAssistant.TargetFramework
+{
+    public class WindowsSdkTargetFrameworkSelectorFilter : ITargetFrameworkSelectorFilter
+    {
+        private const string WindowsSuffix = "-windows";
+
+        public void Process(ITargetFrameworkSelectorFilterState tfm)
+        {
+            if (tfm is null)
+            {
+                throw new System.ArgumentNullException(nameof(tfm));
+            }
+
+            if (TryGetMoniker(tfm, out var result))
+            {
+                tfm.TryUpdate(new TargetFrameworkMoniker(result));
+            }
+        }
+
+        private static bool TryGetMoniker(ITargetFrameworkSelectorFilterState updater, [MaybeNullWhen(false)] out string tfm)
+        {
+            // Projects with Windows Desktop components or a WinExe output type should use a -windows suffix
+            if (updater.Components.HasFlag(ProjectComponents.WindowsDesktop) || updater.Project.OutputType == ProjectOutputType.WinExe)
+            {
+                tfm = $"{updater.AppBase}{WindowsSuffix}";
+
+                if (updater.Components.HasFlag(ProjectComponents.WinRT))
+                {
+                    // TODO: Default to this version to ensure everything is supported.
+                    tfm += "10.0.19041.0";
```

Comment:
I thought of another scenario where these filters won't layer together properly. 

Suppose the same scenario as before (WinExe project with a 5.0 AppBase and 6.0 dependency). If the filters are applied in the opposite order I suggested before, there's still a problem (just a different one):

1. First, the WindowsSdk filter would change Current to 5.0-Windows.
2. But then when the min dependency filter runs it will either change Current to 6.0 or (more likely) leave it as 5.0-Windows (depending on how 5.0-Windows and 6.0 compare). What we *want* is 6.0-Windows.

Ordering issues are creeping in because different filters are changing the TFM in different kinds of ways. We either need to track suffixes like this somewhere in the state so that filters really can execute on the different "parts" of a TFM independently (standard/netcore, AppBase, suffixes) or we need to make sure the filters run in a known good order.

Reply:
I think we could solve that by changing the way `TryUpdate` executes. We aren't looking to compare between 5.0-Windows and 6.0. We're looking to merge the two requirements together.

Can we say that merging 5.0-Windows and 6.0 always produces the same result?

Reply:
Good point. I'm going to look at updating the TryUpdate to be smart about platforms

Reply:
I've added logic for that in TargetFrameworkMonikerComparer